### PR TITLE
Allow multiple uinput devices with the same name

### PR DESCRIPTION
**Pull request by shawnanastasio**
Some peripherals will report the same uinput device name, so update the
get_all_uinput_device_names_to_paths routine to handle this by appending
an incrementing number to the device name in the case of duplicates.

### CODE REVIEW

**Comment by tomitek** [comment]
@shawnanastasio your solution is working - thank you.

**Comment by Thinkpiet** [comment]
This works perfectly! Thanks.

 This PR should be merged to master.

**Comment by RichardFevrier** [comment]
@samvel1024 would you mind to take a look of the PRs please 🙂

File: src/util/util.rs [hunk 1]
```diff
@@ -40,6 +40,26 @@ pub fn get_uinput_device_name(dev_file_path: &String) -> Result<Option<String>>
...
+			map.insert(fixed_name, path);
+			break;
+		} else {
+			i += 1
+		}
+	}
+}
+
 pub fn get_all_uinput_device_names_to_paths() -> Result<HashMap<String, String>> {
 	let paths = fs::read_dir("/dev/input/")?;
```
Comment (mschout) [inline src/util/util.rs:64]: Maybe the `paths` should be sorted in order to ensure that the numbering that is added is deterministic?

Alternatively, I did something simliar [here](https://github.com/samvel1024/kbct/pull/50) by just appending the device number instead

**Pull request merged**